Coroutine.ttslua

```diff
@@ -1,16 +1,28 @@
 ---@class ge_tts__Coroutine
 local Coroutine = {}
 
---- Yields from the current coroutine, which will later be resumed once a condition is met.
+--- Yields from the current coroutine, to be resumed once a condition is met or an optional timeout limit is reached.
 ---@param condition fun(): boolean @Return true when the current coroutine should be resumed.
-function Coroutine.yieldCondition(condition)
+---@param timeout number|nil @Timeout in seconds (optional).
+---@return boolean @Return whether the condition was met or not.
+---@overload fun(condition: fun():boolean):boolean
+function Coroutine.yieldCondition(condition, timeout)
     local co = coroutine.running()
 
-    Wait.condition(function()
-        coroutine.resume(co)
-    end, condition)
+    local timeOutFunc
+    if type(timeout) == "number" then
+        timeOutFunc = function()
+            coroutine.resume(co, false)
+        end
+    end
 
-    coroutine.yield()
+    local function resume()
```

Comment:
I tend to prefer avoiding defining variables if they're just used once. The exceptions are:

* Named constants e.g. Instead of just parsing seemingly arbitrary hard-coded numbers to functions, it's a good idea to have a variable that explains what the variable represents e.g. `MAX_HAND_SIZE` instead of just using `10`.
* A dynamic expression that is either long/complex, or is a value you're likely to want to see in a debugger e.g.
    ```lua
    local somethingDynamic = (aValue > someOtherValue or aNumber == favoriteNumber) and callSomething() or callSomethingElse()
    ```

In this case, it's easy just to pass this function straight to `Wait.condition`. There is an argument to be made for using names when a function take several similar looking arguments, however sane IDEs solve that problem for you:

![Screen Shot 2020-06-10 at 3 32 13 am](https://user-images.githubusercontent.com/482276/84180583-1bd13580-aacb-11ea-913f-9d6b96e5578b.png)

Note the "chips" stating "func" and "conditionFunc", letting you know which is which.